Dockerfile

```diff
@@ -36,16 +37,17 @@ COPY ./log4j.properties src/main/resources/log4j.properties
 
 # build and install, placing the scripts jar back in the target folder
 # where import scripts expect it after cleanup
-RUN mvn -DskipTests clean package \
-	&& unzip portal/target/cbioportal-*.war -d $CATALINA_HOME/webapps/cbioportal \
+RUN mvn -DskipTests clean install \
+	&& unzip portal/target/cbioportal.war -d $CATALINA_HOME/webapps/cbioportal \
 	&& mv scripts/target/scripts-*.jar /root/ \
 	&& mvn clean \
 	&& mkdir scripts/target/ \
 	&& mv /root/scripts-*.jar scripts/target/
 
 # add runtime plumbing to Tomcat config:
 # - make cBioPortal honour db config in portal.properties
-RUN echo 'CATALINA_OPTS="-Dauthenticate=false $CATALINA_OPTS -Ddbconnector=dbcp"' >>$CATALINA_HOME/bin/setenv.sh
+# temporarily add session.service.url here since it does not work in portal.properties
+RUN echo 'CATALINA_OPTS="-Dauthenticate=false $CATALINA_OPTS -Ddbconnector=dbcp -Dsession.service.url=http://cbio-session-service:8080/api/sessions/main_session/"' >>$CATALINA_HOME/bin/setenv.sh
```

Comment:
Will users ever want to override the session service URL, or is this the only URL (protocol and container name) that would work in the context of the rest of this Docker build context?

In the former case, the setting should probably _precede_ the word `$CATALINA_OPTS`, so that it can be overridden by the environment variable like the authentications setting can.

Reply:
I think we have to keep it in the context of the Docker build, the same way as properties in `portal.properties` work. I don't think it makes sense to change this property once the image has been built.

Reply:
That's not right—almost all properties in portal.properties can be overridden without breaking the functionality of the docker image, just like the `authenticate=` setting here.

Reply:
True - We mount `portal.properties` into containers based on the image. I'll move the property to the left of `$CATALINA_OPTS`